grundeinkommensbuero/backend
# Meinbge signup and signin via token

Resolves: https://trello.com/c/wAG9rCIr/755-endpunkt-post-users-external-signup-025
Resolves: https://trello.com/c/y2pkqE7D/784-userinnen-sollen-per-token-eingeloggt-werden-k%C3%B6nnen-2

Endpoint to create user, who signed up for municipal campaign. Endpoint is supposed to be used by MGE via a query token (I'll send you the token via twist). 
Also includes functionality to sign in with a token which was added to the user record in dynamo db beforehand. 

To test:
Check if tests cover most cases, check if code looks good, test endpoint:
POST https://2j0bcp5tr9.execute-api.eu-central-1.amazonaws.com/dev/users/external-signup?token=xxx

To test login via token: after signing up a user via the endpoint get customToken.token from the db and login with that token (you can just type in the e-mail-address in the login screen in the frontend (no email will be sent) and paste in the token as login code --> we will basically do the same automatically after redirecting from Pilotprojekt)

Do you have access to our dynamo db? Maybe we could talk about a setup to test prs tomorrow?
